packages/data-context/src/actions/AppActions.ts

```diff
@@ -48,7 +48,10 @@ export class AppActions {
     let browser
 
     try {
-      browser = (await this.ctx._apis.appApi.ensureAndGetByNameOrPath(browserNameOrPath)) as FoundBrowser | undefined
+      await this.ctx.coreData.app.refreshingBrowsers
+      const foundBrowsers = this.ctx.coreData.app.browsers ? [...this.ctx.coreData.app.browsers] : undefined
+
+      browser = (await this.ctx._apis.appApi.ensureAndGetByNameOrPath(browserNameOrPath, false, foundBrowsers)) as FoundBrowser | undefined
```

Comment:
ensureAndGetByNameOrPath will call the internal getBrowsers utility if no browsers are provided. It seems like we should already have retrieved the browsers in this case or can at least wait until we have to set the active browser. 

The spreading into foundBrowsers is due to the coreData property being readonly; not sure if there's an easier way to cast this?